### PR TITLE
fix: Simplify the build targets

### LINKED ISSUE #53: Cleanup Makefile
We have several Makefiles inside our project. Some of them defined duplicate variables. We should eliminate them.

### PR DESCRIPTION
**Pull request by splasky**
`INCLUDES` variable was defined inside major Makefile. Use $(INCLUDES) for
include all headers inside Ta-endpoint during compiling.

Use `OBJS` for link all needed object files during compiling.  `OBJS`
will search source of c inside directory which include by `SUBDIR`.

Remove Makefile inside utils and connectivity. The objects will be generated
by top of Makefile.

Closes #53

### CODE REVIEW

**Comment by jserv** [comment]
Change the subject to
> Simplify the build targets

File: connectivity/Makefile [hunk 1]
```diff
@@ -1,13 +1,10 @@
...
+conn_http.o: conn_http.c
 	@echo Compiling $@ ...
 	$(CC) -v -c $(CFLAGS) $(INCLUDES) -MMD -MF conn_http.c.d -o $@ $<
 
 clean:
 	rm -f *.o *.c.d
 
--include conn_http.c.d
\ No newline at end of file
+-include conn_http.c.d
```
Comment (jserv) [inline connectivity/Makefile]: Set via variable `$(deps)`, which should be followed by `conn_http.o`.

File: connectivity/Makefile [hunk 2]
```diff
@@ -1,13 +1,11 @@
-INCLUDES := -I$(THIRD_PARTY_PATH)/mbedtls/include -I$(THIRD_PARTY_PATH)/http-parser -I$(ROOT_DIR)/utils
-
 all: conn_http.o
+deps := conn_http.c.d
```
Comment (jserv) [inline connectivity/Makefile]: Don't do this.
Use something like `deps += $(OBJS:%.o=%.o.d)`

File: connectivity/Makefile [hunk 3]
```diff
@@ -1,13 +1,11 @@
...
 
-.NOTPARALLEL:
-connectivity/conn_http.o: conn_http.c
+conn_http.o: conn_http.c
 	@echo Compiling $@ ...
-	$(CC) -v -c $(CFLAGS) $(INCLUDES) -MMD -MF conn_http.c.d -o $@ $<
+	$(CC) -v -c $(CFLAGS) $(INCLUDES) -MMD -MF $(deps) -o $@ $<
```
Comment (jserv) [inline connectivity/Makefile]: Change `$(deps)` to `$@.d`

**Comment by jserv** [comment]
Did you verify parallel build?

**Comment by splasky** [comment]
I tested with command `make -j$(nproc)`. It only generated error message from #61. After fix it. It works well.

**Comment by jserv** [comment]
The git commit message should mention the removal of `Makefile` entries in directory `utils` and `connectivity`.

**Pull request merged**